### PR TITLE
Replace PrintF with Console.WriteLine and cleanups

### PR DESCRIPTION
**Pull request by am11**
The three usages of these vararg wrapper functions are using fixed number of arguments (1).

### CODE REVIEW

**Comment by am11** [comment]
For testing comparisons of PAL snprintf which uses libc's vsnprintf (master) with libc's snprintf directly (PR); a gist attempts to make sure there are no differences before and after: [gist](https://gist.github.com/am11/e1a1bbdab077e28f4e098d5089421867) | [usage](https://gist.github.com/am11/e1a1bbdab077e28f4e098d5089421867#gistcomment-3045383)

**Comment by stephentoub** [comment]
Thanks for the PR, but I'm not seeing the benefit of this change.  We explicitly do not want to P/Invoke directly into libc from corefx, and only ever do so in test code.  Can you please explain why this is a beneficial change?

**Comment by am11** [comment]
Sorry I did not realize that I am using antipattern by P/Invoking libc directly. I should have looked more carefully. :)

I was reading the C code in `pal_string.h` and thought that `snprintf` and `printf` could just be alias to C library functions with `PAL` prefix; without re-implementing snprintf using vsnprintf (which felt unnecessary). For safe strings, `snprintf_s` and `printf_s` etc. could be used from C library.

After looking at the usages, I found that printf shim is only ever used in one place, under `#if DEBUG` to print a debug string on stdout (return integer is ignored), and snprintf four places in a method. Therefore I went ahead with `internal` to `private` cleanup.

If it makes sense to redefine PrintF/SNPrintF as simple alias to C lib function in pal_string.c (keeping the shim layer intact),  I can re-purpose this PR.

**Comment by stephentoub** [comment]
Thanks.  I don't think we should change SNPrintF.  Since as you say PrintF is only ever used in debug and in one place, we could delete that entirely and just use Console.WriteLine at the call sites; doing so would require adding a reference to System.Console from several projects, and we'd want to only do so in DEBUG.

File: src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs [hunk 1]
```diff
@@ -21,8 +21,8 @@ internal sealed class SafeX509Handle : SafeHandle
         {
             if (s_captureTrace)
             {
-                Interop.Sys.PrintF(
-                    "%s\n\n",
+                Console.WriteLine(
+                    "{0}\n\n",
```
Comment (stephentoub) [inline src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs]: This first argument to WriteLine can be deleted entirely.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Comment by am11** [comment]
@stephentoub, thanks.

During the testing, it seemed no current test case exercises this code path leading to vsnprintf call: https://github.com/dotnet/corefx/blob/99ea2b5aebf7361a5baa638c978cb6214fca5217/src/System.Console/src/System/TermInfo.cs#L871 including the `System.Console.Manual.Tests`. I wasn't able to figure out a way either to invoke it from public methods via a unit test or a manual test. Therefore, I isolated this method in a gist mentioned above. That test shows that this method:

* works fine when format is `"string with wrong format specifier %d"` and arg is `"test"`.
* throws `InvalidCastException`, when format is `"test float value %d"` and arg is float `0.8`.
* terminates with `SIGSEGV`, when format is `"int with wrong format specifier %s"` and arg is int.

IMO, we can try to cover this method in unit tests. Unless it is dead code, in which case it can be removed as well. :)

**Comment by stephentoub** [comment]
It isn't dead. It's used by some terminfo files that include specific formats.

> terminates with SIGSEGV, when format is "int with wrong format specifier %s" and arg is int.

That's printf itself failing. If you give it bad inputs, such as telling it to print the string at some invalid address, it will likely fail.

**Pull request merged**